src/Juicer.cpp

```diff
@@ -2203,7 +2205,7 @@ JuicerEndianness_t Juicer::getEndianness()
 
 			ident_buffer = elf_hdr_32->e_ident;
 
-			if(ident_buffer[EI_DATA] == 0)
+			if(ident_buffer[EI_DATA] == 2)
```

Comment:
libelf has a macro for this called `ELFDATA2MSB`. I actually used it on a branch that has not been merged yet on  94dee96b7ed76c2a20b26a9ed9792c58885973f8 commit.